packages/core/src/service/services/product.service.ts

```diff
@@ -228,6 +228,8 @@ export class ProductService {
             channelId: input.channelId,
             priceFactor: input.priceFactor,
         });
+        const assetIds: ID[] = ([] as ID[]).concat(...productsWithVariants.map(p => p.assets.map(a => a.id)));
```

Comment:
I think it would make sense to apply the `unique()` helper function here since multiple products may have the exact same assets.